src/pages/index.js

```diff
@@ -0,0 +1,365 @@
+      avatarImage.src = userInfo.avatar;
+    }
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+
+    cards.forEach((card) => renderCard(card));
+  })
+  .catch((err) => {
+    console.error("Error fetching app info:", err);
+    alert("Could not load profile or cards. Please try again later.");
+  });
+
+profileEditButton.addEventListener("click", () => {
+  resetValidation(editFormElement, settings);
+
+  editModalNameInput.value = profileName.textContent;
+  editModalDescription.value = profileDescription.textContent;
+
+  openModal(editModal);
+});
+
+const editModalCloseBtn = editModal.querySelector(".modal__close-button");
+editModalCloseBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+const loadingModal = document.querySelector("#loading-modal");
+
+function showLoadingModal(loadingText = "Saving...") {
+  const loadingTextElement = loadingModal.querySelector(".modal__loading-text");
+  loadingTextElement.textContent = loadingText;
+  openModal(loadingModal);
+}
+
+function hideLoadingModal() {
+  closeModal(loadingModal);
+}
+
+editFormElement.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  showLoadingModal("Saving profile..."); // Show loading modal
+
+  const originalButtonText = editSubmitButton.textContent;
+  editSubmitButton.textContent = "Saving...";
+  editSubmitButton.disabled = true;
+
+  const updatedName = editModalNameInput.value;
+  const updatedAbout = editModalDescription.value;
+
+  api
+    .editUserInfo({ name: updatedName, about: updatedAbout })
+    .then((updatedUserInfo) => {
+      profileName.textContent = updatedUserInfo.name;
+      profileDescription.textContent = updatedUserInfo.about;
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+      alert("Could not update profile. Please try again later.");
+    })
+    .finally(() => {
+      hideLoadingModal(); // Hide loading modal
+      editSubmitButton.textContent = originalButtonText;
+      editSubmitButton.disabled = false;
+    });
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalOnEscape);
+  modal.addEventListener("mousedown", closeModalOnOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalOnEscape);
+  modal.removeEventListener("mousedown", closeModalOnOverlay);
+}
+
+function closeModalOnEscape(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function closeModalOnOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function disableButton(buttonElement, config) {
+  buttonElement.classList.add(config.inactiveButtonClass);
+  buttonElement.disabled = true;
```

Comment:
Please, delete `disableButton` from `index.js` because you have it inside `validation.js`. You should **import** it from there